hfile.c

```diff
@@ -506,12 +518,13 @@ static off_t fd_seek(hFILE *fpv, off_t offset, int whence)
 
 static int fd_flush(hFILE *fpv)
 {
-    hFILE_fd *fp = (hFILE_fd *) fpv;
-    int ret;
+    int ret = 0;
     do {
 #ifdef HAVE_FDATASYNC
+        hFILE_fd *fp = (hFILE_fd *) fpv;
         ret = fdatasync(fp->fd);
-#else
+#elif defined(HAVE_FSYNC)
+        hFILE_fd *fp = (hFILE_fd *) fpv;
         ret = fsync(fp->fd);
```

Comment:
A windows implementation of fsync makes sense, but it probably ought to be surrounded by a ```#ifndef HAVE_FSYNC``` check and appropriate autoconf magic.  Indeed hts_os.h sounds like the right spot for it too.

I'm not such a fan of ```#define HAVE_STRUCT_STAT_ST_BLKSIZE``` appearing mid-way through a file as the use of it can lead people to wrong conclusions (that it's in a header file somewhere and can be used elsewhere), given the uses of these are several pages below their defines.